lib/view_component/base.rb

```diff
@@ -104,11 +104,16 @@ def render_in(view_context, &block)
       before_render
 
       if render?
-        # Avoid allocating new string when output_postamble is blank
-        if output_postamble.blank?
-          safe_render_template_for(@__vc_variant).to_s
+        # Avoid allocating new string when output_preamble and output_postamble are blank
+        rendered_template = safe_render_template_for(@__vc_variant).to_s
+
+        if output_preamble.blank? && output_postamble.blank?
+          rendered_template
         else
-          safe_render_template_for(@__vc_variant).to_s + safe_output_postamble
+          preamble = output_preamble.blank? ? "" : safe_output_preamble
+          postamble = output_postamble.blank? ? "" : safe_output_postamble
+
+          preamble + rendered_template + postamble
```

Comment:
`#maybe_escape_html` already checks for blankness, so I think this code can be simplified a bit:

```suggestion
          safe_output_preamble + rendered_template + safe_output_postamble
```

Reply:
Hey @camertron, I accepted your suggestion, but it looks like the build then started failing on the Primer tests. Any idea why this change could have done that?

Reply:
Yeah, not sure, will look into it this afternoon 👍